docs/index.md

```diff
@@ -1,9 +1,80 @@
 # Xarray Custom Indexes Gallery
 
-## What are Indexes?
+## Background
+
+Xarray's data model was initially heavily inspired from the
+[NetCDF](https://www.unidata.ucar.edu/software/netcdf/) file format, making it
+well suited for working with n-dimensional, rectilinear gridded datasets
+commonly found in scientific data analysis, especially in the geosciences.
+
+```{figure} _static/figs/xarray-dataset-diagram-legacy.png
+---
+alt: Xarray data model
+align: center
+width: 500px
+class: dark-light
+---
+An illustration of the Xarray data model.
+```
+
+Over the years, Xarray has been used in an increasing number of domains as a
+convenient, general-purpose Python library for handling n-dimensional labeled
+arrays. Xarray's data structures are now reused for representing a wide range of
+datasets including sparse data, curvilinear or irregular grids, staggered grids,
+discrete global grids, geospatial raster and vector data cubes.
+
+```{figure} _static/figs/xarray-dataset-diagram-new.png
+---
+alt: Xarray data model
+align: center
+width: 500px
+class: dark-light
+---
+A better illustration of the variety of Xarray datasets in the wild.
+```
+
+## What is an Xarray index?
+
+In a (relational) database, an index is a specific and opaque structure that
+allows fast data lookup and retrieval. Likewise, a {py:class}`pandas.Index`
+object can be used to efficiently select rows of a {py:class}`pandas.DataFrame`
+by one or more labels.
+
+Until recently Xarray exclusively relied on {py:class}`pandas.Index` to allow
+fast label-based selection of n-dimensional data via the concept of
+{term}`"dimension" coordinates <xarray:Dimension coordinate>`. This approach
+worked very well for rectilinear gridded datasets but quickly reached its
+limitations when considering other kinds of datasets.
+
+While Xarray still follows the same approach as its default behavior, it has
```

Comment:
I think this can go after the next paragraph which describes what an Xarray index does.